client/app/components/compose.coffee

```diff
@@ -50,21 +50,129 @@ module.exports = Compose = React.createClass
     getDefaultProps: ->
         layout: 'full'
 
+    getInitialState: ->
+        MessageUtils.createBasicMessage @props
+
+    isNew: ->
+        not @state.conversationID
 
     shouldComponentUpdate: (nextProps, nextState) ->
-        return not(_.isEqual(nextState, @state)) or
-            not (_.isEqual(nextProps, @props))
+        !!nextProps.accounts
 
+    componentWillReceiveProps: (nextProps) ->
+        if nextProps.message
```

Comment:
You can probably avoid having to manage this situation (the compose component is injected with another message) by using smart key (compose-#{message.id} ?) attribute in panel.coffee